src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataColumn.java

```diff
@@ -45,4 +45,35 @@ public String getColumnName() {
     public int getColumnType() {
         return javaSqlType;
     }
+
+    @Override
+    public int hashCode() {
+        int hash = 7;
+        hash = 31 * hash + javaSqlType;
+        hash = 31 * hash + precision;
+        hash = 31 * hash + scale;
+        hash = 31 * hash + numberOfDigitsIntegerPart;
+        hash = 31 * hash + (null != columnName ? columnName.hashCode() : 0);
+        return hash;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (null != object && object.getClass() == SQLServerDataColumn.class) {
+            SQLServerDataColumn aSQLServerDataColumn = (SQLServerDataColumn) object;
+            if (hashCode() == aSQLServerDataColumn.hashCode()) {
+                // Compare objects to avoid collision
+                return ((null == columnName && null == aSQLServerDataColumn.columnName

```

Comment:
This will probably throw a NPE when `this.columnName` is null, but `aSQLServerDataColumn.columnName` is not null. Better would be `java.util.Objects.equals(columnName, aSQLServerDataColumn.columnName)` or `(columnName == null ? aSQLServerDataColumn.columnName == null : columnName.equals(aSQLServerDataColumn.columnName))`

Reply:
Hi @TheConstructor,
Good catch 👍 I created #1168  with your suggestion.